src/disassemble.cpp

```diff
@@ -308,57 +269,47 @@ void REHex::Disassemble::update()
 
 void REHex::Disassemble::reinit_disassembler()
 {
-	const char *triple = arch_list[ arch->GetSelection() ].triple;
+	const CSArchitecture& desc = arch_list[ arch->GetSelection() ];
 	
-	std::call_once(g_Initialize_disassembler, Initialize_disassembler);
-	if(disassembler != NULL)
+	if(disassembler != 0)
 	{
-		LLVMDisasmDispose(disassembler);
-		disassembler = NULL;
+		cs_close(&disassembler);
 	}
 	
-	disassembler = LLVMCreateDisasm(triple, NULL, 0, NULL, NULL);
-	if(disassembler == NULL)
+	cs_err error = cs_open(desc.arch, desc.mode, &disassembler);
+	if(error != CS_ERR_OK)
 	{
 		/* TODO: Report error */
 		return;
 	}
 	
-	/* Use Intel assembly syntax. */
-	LLVMSetDisasmOptions(disassembler, LLVMDisassembler_Option_AsmPrinterVariant);
+	/* Enable SKIPDATA mode */
+	/* This ensures the diassembler continues on unknown data, displaying it as '.byte 00' */
+	cs_option(disassembler, CS_OPT_SKIPDATA, CS_OPT_ON);
```

Comment:
I think this option should probably be off - if you look in `REHex::Disassemble::update()` you'll see it attempts to align the disassembly with the cursor, or to the nearest point that produces valid disassembly, so you can click near an instruction and on certain architectures (almost any garbage looks like valid x86 machine code), it (might) find the instruction boundaries/alignment correctly.

Are there any situations you can think of where getting disassembly which is just full of "data" would be preferable?

Reply:
Usually (at least on x64) it is just a one-byte pad:

![image](https://user-images.githubusercontent.com/610685/84602026-c1e2be00-ae84-11ea-9ff9-ca95cdd24462.png)

But if you want this option disabled (which makes it behave more like it did before with llvm!) that is totally fine with me.